engine-comparison/dispatcher.sh

```diff
@@ -7,60 +7,84 @@
 
 build_engine() {
   # [[ ! -e ~/$FENGINE_CONFIGS_DIR/$1 ]] && echo "cant do" && exit 1
-
   # All of this code will probably be in a separate script
 
   FENGINE_CONFIG=$1
 
   echo "Building $FENGINE_CONFIG"
-  export ENGINE_DIR=$WORK/fuzz-engines/${FENGINE_CONFIG}
+  FENGINE_NAME=$(basename $FENGINE_CONFIG)
+  export ENGINE_DIR=$WORK/fuzz-engines/${FENGINE_NAME}
   rm -rf $ENGINE_DIR
   mkdir $ENGINE_DIR
 
   # Fuzzing activities go here
-
 }
 
 build_benchmark_using() {
   BENCHMARK=$1
   FENGINE_CONFIG=$2
+  THIS_BENCHMARK=$3
 
+  BUILDING_DIR=$WORK/BUILD-${THIS_BENCHMARK}
   echo "Filling $BUILDING_DIR"
-  BUILDING_DIR=$WORK/BUILD-${BENCHMARK}-WITH-${FENGINE_CONFIG}
   rm -rf $BUILDING_DIR
   mkdir $BUILDING_DIR
 
   # Do fuzzing things here
   # [[ ! -e ~/FTS/$BENCHMARK/build.sh ]] && echo "cant build" && exit 1
   # $WORK/FTS/$BENCHMARK/build.sh $FUZZING_ENGINE
 
-  export SEND_DIR=$WORK/SEND-${BENCHMARK}-WITH-${FENGINE_CONFIG}
+  export SEND_DIR=$WORK/SEND-${THIS_BENCHMARK}
   rm -rf $SEND_DIR
   mkdir $SEND_DIR
+  echo "Test file" > $SEND_DIR/example.txt
   # Then construct the directory to send here, e.g.
   # cp ${BENCHMARK}-${FUZZING_ENGINE} $SEND_DIR
   # copy seeds, afl-fuzz, etc
 
   rm -rf $BUILDING_DIR
 }
 
-BASE_INSTANCE_NAME="FTS-RUNNER"
+handle_benchmark() {
+  BENCHMARK=$1
+  FENGINE_CONFIG=$2
+  THIS_BENCHMARK=${BENCHMARK}-WITH-$(basename {FENGINE_CONFIG}) # Just for convenience
+  build_benchmark_using $BENCHMARK $FENGINE_CONFIG $THIS_BENCHMARK
 
-for FENGINE_CONFIG in $(find $WORK/fengine-configs); do
-  build_engine $FENGINE_CONFIG
-  for BENCHMARK in $ALL_BENCHMARKS; do
-    # n.b. this requires each config file to have a different name
-    build_benchmark_using $BENCHMARK $FENGINE_CONFIG
+  INSTANCE_NAME=FTS-RUNNER-${THIS_BENCHMARK}
+  gcloud compute instances create $INSTANCE_NAME --zone=$GCLOUD_ZONE
+  # TODO robust ssh here (equiv scp)
+  gcloud compute scp $WORK/SEND-${THIS_BENCHMARK}/ ${INSTANCE_NAME}:~/input --recurse --zone=$GCLOUD_ZONE
```

Comment:
Regarding this and the above comment, concurrency has thus far been instable. I've been trying to use SEND_DIR here, and would like to, but it would require at least rearranging some of this script.